### PR TITLE
Swagger documentation

### PR DESCRIPTION
**Pull request by adelowo**
See https://github.com/TykTechnologies/product/issues/88

- [x] Definition of swagger models and responses
- [x] CI build step to validate the schema

### CODE REVIEW

**Comment by adelowo** [comment]
Right now, the spec will be validated in CI. But in the long run, we might want to push the file 
directly to GitHub from Travis but for now, `.utils/ci-swagger.sh` can be run and the changes in the
`gateway-swagger.yaml` can be committed alongside other change - CI will validate it again though.

File: swagger.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/TykTechnologies/tyk/user"
+)
+
+// parameterBodies
+// swagger:response parameterBodies
+type swaggerParameterBodies struct {
```
Comment (buger) [inline swagger.go]: What it is? and why it is needed?

**Comment by buger** [comment]
Nice!

As mentioned previously our swagger consist of 2 parts, manual part and auto-generated part.
So next step here, will be taking swagger.yaml from here `https://github.com/TykTechnologies/tyk-docs/blob/db26f221108604da80d7cdf4a5469b83e722c1c7/tyk-docs/static/others/gateway-swagger.yaml` and replacing its `components` object, part, with auto-generated stuff from your script: so we will have manually defined endpoints and automatic model definitions.

File: swagger.go (hunk 1 above)
Comment (adelowo) [reply swagger.go]: Without this, `go-swagger` won't generate definitions for `apiStatusMessage` and the rest. Thus the swagger spec will be invalid as `"$ref": "#/definitions/APIDefinition"` for example won't work. 

It is more like a way to actually trick `go-swagger` to generating them as definitions in the output file

**Comment by adelowo** [comment]
I might need to make a few updates to this later on today

**Comment by buger** [comment]
The final change is to move the final swagger file from this PR to here https://github.com/TykTechnologies/tyk-docs/pull/560/files, and modify the script, so it updates definitions section.

**Pull request merged**